file.go

```diff
@@ -178,3 +181,57 @@ func (k *fileKeyring) Keys() ([]string, error) {
 
 	return keys, nil
 }
+
+func (k *fileKeyring) getNewPassword() error {
+	dir, err := k.resolveDir()
+	if err != nil {
+		return err
+	}
+
+	if k.newPassword == "" {
+		pwd, err := k.newPasswordFunc(fmt.Sprintf("Enter new passphrase for %q", dir))
+		if err != nil {
+			return err
+		}
+		k.newPassword = pwd
```

Comment:
I suggest you avoid storing the new password in the struct. You can handle everything directly within the Changepw method

Reply:
You mean to pass in the new password by argument to the function?

Reply:
no need to pass the new pswd as an argument, I guess you can do it in place inside the func